velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,365 @@
+import com.google.gson.JsonParser;
+import com.velocitypowered.api.command.CommandSource;
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.api.proxy.ProxyServer;
+import me.leoko.advancedban.AbstractMethodInterface;
+import me.leoko.advancedban.Universal;
+import me.leoko.advancedban.manager.PunishmentManager;
+import me.leoko.advancedban.manager.UUIDManager;
+import me.leoko.advancedban.utils.Permissionable;
+import me.leoko.advancedban.utils.Punishment;
+import me.leoko.advancedban.utils.tabcompletion.TabCompleter;
+import me.leoko.advancedban.velocity.command.CommandRecieverVelocity;
+import me.leoko.advancedban.velocity.event.PunishmentEvent;
+import me.leoko.advancedban.velocity.event.RevokePunishmentEvent;
+import me.leoko.advancedban.velocity.utils.LuckPermsOfflineUser;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.objectmapping.ObjectMappingException;
+import ninja.leaping.configurate.yaml.YAMLConfigurationLoader;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class VelocityMethods extends AbstractMethodInterface<ConfigurationNode> {
+
+  private final ProxyServer server;
+  private final Path dataDirectory;
+  private final Logger logger;
+
+  private boolean luckPermsSupport;
+
+  public VelocityMethods(Path dataDirectory,
+                         final @NonNull ProxyServer server,
+                         Logger logger) {
+    super(dataDirectory);
+    this.dataDirectory = dataDirectory;
+    this.server = server;
+    this.logger = logger;
+
+    // LuckPerms check stuff here
+
+  }
+
+
+  @Override
+  protected ConfigurationNode loadConfiguration(Path configPath) throws IOException {
+      YAMLConfigurationLoader loader = YAMLConfigurationLoader.builder().setPath(configPath).build();
+      return loader.load();
+  }
+
+  @Override
+  public String getFromUrlJson(String url, String key) {
+    try {
+      HttpURLConnection request = (HttpURLConnection) new URL(url).openConnection();
+      request.connect();
+
+      JsonParser jp = new JsonParser();
+      JsonObject json = (JsonObject) jp.parse(new InputStreamReader(request.getInputStream()));
+
+      String[] keys = key.split("\\|");
+      for (int i = 0; i < keys.length - 1; i++) {
+        json = json.getAsJsonObject(keys[i]);
+      }
+
+      return json.get(keys[keys.length - 1]).toString().replaceAll("\"", "");
+
+    } catch (Exception exc) {
+      return null;
+    }
+  }
+
+  @Override
+  public String getVersion() {
+    return server.getVersion().toString();
```

Comment:
Using ProxyVersion#toString is not right. getVersion needs to be implemented to return the version of AdvancedBan.